crates/ide_completion/src/render/struct_literal.rs

```diff
@@ -0,0 +1,131 @@
+//! Renderer for `struct` literal.
+
+use hir::{db::HirDatabase, HasAttrs, HasVisibility, Name, StructKind};
+use ide_db::helpers::SnippetCap;
+use itertools::Itertools;
+
+use crate::{item::CompletionKind, render::RenderContext, CompletionItem, CompletionItemKind};
+
+pub(crate) fn render_struct_literal(
+    ctx: RenderContext<'_>,
+    strukt: hir::Struct,
+    local_name: Option<Name>,
+) -> Option<CompletionItem> {
+    let _p = profile::span("render_struct_literal");
+
+    let fields = strukt.fields(ctx.db());
+    let (visible_fields, fields_omitted) = visible_fields(&ctx, &fields, strukt)?;
+
+    if fields_omitted {
+        // If some fields are private you can't make `struct` literal.
+        return None;
+    }
+
+    let name = local_name.unwrap_or_else(|| strukt.name(ctx.db())).to_string();
+    let literal =
+        render_literal(&ctx, &name, strukt.kind(ctx.db()), &visible_fields, fields_omitted)?;
```

Comment:
There is no point in threading `fields_omitted` through the calls here as we are always returning early in case its true. Should simplify things a bit.